kolibri/core/assets/src/views/PaginatedListContainer.vue

```diff
@@ -0,0 +1,161 @@
+<template>
+
+  <div>
+    <KGrid>
+      <KGridItem sizes="3, 3, 3">
+        <slot name="otherFilter"></slot>
+      </KGridItem>
+      <KGridItem sizes="9, 9, 9" align="right" class="text-filter">
+        <KFilterTextbox
+          v-model.trim="filterInput"
+          :placeholder="filterPlaceholder"
+          @input="pageNum = 1"
+        />
+      </KGridItem>
+    </KGrid>
+
+    <div>
+      <slot v-bind="{items: visibleFilteredItems, filterInput}"></slot>
+    </div>
+
+    <nav>
+      <span dir="auto" class="pagination-label">
+        {{ $tr('pagination', { visibleStartRange, visibleEndRange, numFilteredItems }) }}
+      </span>
+      <UiIconButton
+        type="primary"
+        :ariaLabel="$tr('previousResults')"
+        :disabled="pageNum === 1"
+        size="small"
+        class="pagination-button"
+        @click="changePage(-1)"
+      >
+        <KIcon icon="back" style="position: relative; top: -1px;" />
+      </UiIconButton>
+      <UiIconButton
+        type="primary"
+        :ariaLabel="$tr('nextResults')"
+        :disabled="pageNum === 0 || pageNum === numPages"
+        size="small"
+        class="pagination-button"
+        @click="changePage(+1)"
+      >
+        <KIcon icon="forward" style="position: relative; top: -1px;" />
+      </UiIconButton>
+    </nav>
+  </div>
+
+</template>
+
+
+<script>
+
+  import UiIconButton from 'kolibri.coreVue.components.UiIconButton';
+  import KFilterTextbox from 'kolibri.coreVue.components.KFilterTextbox';
+  import KIcon from 'kolibri.coreVue.components.KIcon';
+  import KGrid from 'kolibri.coreVue.components.KGrid';
+  import KGridItem from 'kolibri.coreVue.components.KGridItem';
+
+  export default {
+    name: 'PaginatedListContainer',
+    components: {
+      KIcon,
+      UiIconButton,
+      KFilterTextbox,
+      KGrid,
+      KGridItem,
+    },
+    props: {
+      // The entire list of items
+      items: {
+        type: Array,
+        required: true,
+      },
+      filterFunction: {
+        type: Function,
+      },
+      filterPlaceholder: {
+        type: String,
+        required: true,
+      },
+      otherFilters: {
+        type: Object,
+        required: false,
+      },
+    },
+    data() {
+      return {
+        filterInput: '',
+        pageNum: 1,
+        perPage: 10,
```

Comment:
My sense is that 10 is way too low .. maybe we should bump this up to 30 or so?

Reply:
sounds good to me